microcosm_flask/conventions/upload.py

```diff
@@ -0,0 +1,44 @@
+"""
+Conventions for file upload.
+
+"""
+from flask import request
+from microcosm_flask.conventions.base import Convention
+from microcosm_flask.operations import Operation
+from werkzeug.exceptions import BadRequest
+
+
+class UploadConvention(Convention):
+
+    def configure_upload(self, ns, definition):
+        """
+        Register an upload endpoint.
+
+        The definition's func should be an upload function, which must:
+        - accept kwargs for path data
+        - accept a dict of files
+
+        :param ns: the namespace
+        :param definition: the endpoint definition
+
+        """
+        @self.graph.route(ns.collection_path, Operation.Upload, ns)
+        def upload(**path_data):
+            files = request.files
+
+            if not files:
+                raise BadRequest("No files uploaded!")
+
+            definition.func(files, **path_data)
```

Comment:
@jessemyers FYI there is a bit of nasty boilerplate further required to actually use an uploaded file - see [this](https://github.com/globality-corp/escolastica/blob/develop/escolastica/utils/file_utils.py#L11). Would it make sense to move this 'upload_tmp_file' helper to microcosm-flask somewhere?